test/gyppies-test.js

```diff
@@ -24,9 +30,12 @@ function build(name) {
     const spawnOpts = { stdio: stdio, cwd: folder };
 
     const argvFile = path.join(folder, 'argv.json');
-    const argv = fs.existsSync(argvFile) ? require(argvFile) : [];
+    let argv = fs.existsSync(argvFile) ? require(argvFile) : [];
+    argv = [ gyp ].concat(argv);
+    if (process.env.running_under_istanbul)
+      argv = istanbulArgs.concat(argv);
 
-    let p = spawnSync(process.execPath, [ gyp ].concat(argv), spawnOpts);
+    let p = spawnSync(process.execPath, argv, spawnOpts);
```

Comment:
Thoughts on whether to tweak this to:

    const argvFile = path.join(folder, 'argv.json');
    let argv = fs.existsSync(argvFile) ? require(argvFile) : [];
    if (process.env.running_under_istanbul)
      argv = istanbulArgs.concat(argv);

    let p = spawnSync(process.execPath, [ gyp ].concat(argv), spawnOpts);

(Kinda just puts it back more in line with the existing code (which is my normal preference); happy to, or happy to not :-) )